### PR TITLE
feat(testrunner): allow external reporters

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -83,6 +90,11 @@ jobs:
       with:
         name: ${{ matrix.browser }}-mac-testrun.log
         path: testrun.log
+    - uses: actions/upload-artifact@v1
```
Comment (dgozman) [inline .github/workflows/tests.yml:93]: Why not put everything into `test-results` folder? E.g. `test-results/results.json`, `test-results/testrun.log`.

File: test-runner/src/reporters/json.ts [hunk 2]
```diff
@@ -16,15 +16,20 @@
...
+class JSONReporter extends BaseReporter {
   onEnd() {
     super.onEnd();
     const result = {
       config: this.config,
       suites: this.suite.suites.map(suite => this._serializeSuite(suite)).filter(s => s)
     };
-    console.log(JSON.stringify(result, undefined, 2));
+    const report = JSON.stringify(result, undefined, 2);
+    if (process.env.PWRUNNER_JSON_REPORT)
```
Comment (dgozman) [inline test-runner/src/reporters/json.ts:29]: Mocha has `--reporter-option`: https://mochajs.org/#-reporter-option-option-o-option-reporter-options-option

**Review by dgozman — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (pavelfeldman) [reply .github/workflows/tests.yml:93]: Good call.

File: test-runner/src/reporters/json.ts (hunk 2 above)
Comment (pavelfeldman) [reply test-runner/src/reporters/json.ts:29]: Yeah, also good call. I'll do both.

**Pull request merged**